lib/rex/post/meterpreter/ui/console/command_dispatcher/kiwi.rb

```diff
@@ -643,6 +644,54 @@ def scrape_passwords(provider, method, args)
     return true
   end
 
+  def report_creds(type, rows)
+    user, domain, pass = rows
+    case type
+    when :msv
+      user = rows[0]
+      domain = rows[1]
+      lm_hash = rows[2].downcase
+      nt_hash = rows[3].downcase
+      return if (lm_hash.eql?('aad3b435b51404eeaad3b435b51404ee') && nt_hash.eql?('31d6cfe0d16ae931b73c59d7e0c089c0'))
+
+      pass = "#{lm_hash}:#{nt_hash}"
+    when :wdigest, :kerberos
+      user = rows[0]
+      domain = rows[1]
+      pass = rows[2]
+    end
+    return if (user.empty? || pass.eql?('(null)'))
```

Comment:
Technically speaking we have more than just `msv`, `wdigest`, and `kerberos` here. If you include all potential options, we also have `livessp`, `ssp`, and `tspkg` here which need to be handled appropriately.

Reply:
From https://ivanitlearning.wordpress.com/2019/09/07/mimikatz-and-password-dumps/ it looks like we could just do this to add support for `tspkg`:

```suggestion
    case type
    when :msv
      user = rows[0]
      domain = rows[1]
      lm_hash = rows[2].downcase
      nt_hash = rows[3].downcase
      return if (lm_hash.eql?('aad3b435b51404eeaad3b435b51404ee') && nt_hash.eql?('31d6cfe0d16ae931b73c59d7e0c089c0'))

      pass = "#{lm_hash}:#{nt_hash}"
    when :wdigest, :kerberos, :tspkg
      user = rows[0]
      domain = rows[1]
      pass = rows[2]
    end
    return if (user.empty? || pass.eql?('(null)'))
```
Looking into `livessp` and `ssp` some more to get more details on how those work.

Reply:
Alright looks like `ssp` is also in plaintext according to https://github.com/gentilkiwi/mimikatz/issues/293 unless Credential Guard is enabled, so we could update the above line to include `:ssp` as well.

Reply:
@cn-kali-team Marking this as unresolved for now whilst we figure out how to also handle `livessp` and `ssp` appropriately. Appreciate you fixing the `tspkg` issue though!

Reply:
Okay looking at the following code from another module we have, it seems like `livessp`, `ssp`, `kerberos`, and `tspkg` all have a username, domain, and password component so we should just be able to update this code with those strings. https://github.com/rapid7/metasploit-framework/blob/85a9accbeec063f52406621f124a7f82ad63dcba/modules/post/windows/gather/credentials/sso.rb#L71-L85.

Reply:
Resolved now with https://github.com/rapid7/metasploit-framework/pull/14432/commits/be9291a85fc642e5ebd346b2791f98040502298a